resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/adaptive/AdaptiveBulkhead.java

```diff
@@ -0,0 +1,575 @@
+			long start = 0;
+			boolean isFailed=false;
+			bulkhead.acquirePermission();
+			try {
+				start = Instant.now().toEpochMilli();
+				return function.apply(t);
+			} catch (Exception e) {
+				isFailed=true;
+				bulkhead.onError( start,TimeUnit.MILLISECONDS, e);
+				throw e;
+			} finally {
+				if (start != 0 && !isFailed) {
+					bulkhead.onSuccess(start, TimeUnit.MILLISECONDS);
+				}
+			}
+		};
+	}
+
+	/**
+	 * Returns a function which is decorated by a bulkhead.
+	 *
+	 * @param bulkhead the bulkhead
+	 * @param function the original function
+	 * @param <T>      the type of the input to the function
+	 * @param <R>      the type of the result of the function
+	 * @return a function which is decorated by a bulkhead.
+	 */
+	static <T, R> CheckedFunction1<T, R> decorateCheckedFunction(AdaptiveBulkhead bulkhead, CheckedFunction1<T, R> function) {
+		return (T t) -> {
+			long start = 0;
+			boolean isFailed=false;
+			bulkhead.acquirePermission();
+			try {
+				start = Instant.now().toEpochMilli();
+				return function.apply(t);
+			} catch (Exception e) {
+				isFailed=true;
+				bulkhead.onError( start,TimeUnit.MILLISECONDS, e);
+				throw e;
+			} finally {
+				if(start != 0 && !isFailed) {
+					bulkhead.onSuccess(start, TimeUnit.MILLISECONDS);
+				}
+			}
+		};
+	}
+
+
+
+	/**
+	 * Create a Bulkhead with a default configuration.
+	 *
+	 * @param name the name of the bulkhead
+	 * @return a Bulkhead instance
+	 */
+	static AdaptiveBulkhead ofDefaults(String name) {
+		return AdaptiveLimitBulkhead.factory().createAdaptiveLimitBulkhead(name, AdaptiveBulkheadConfig.ofDefaults());
+	}
+
+
+	/**
+	 * Creates a bulkhead with a custom configuration and custom limiter
+	 *
+	 * @param name         the name of the bulkhead
+	 * @param config       a custom BulkheadConfig configuration
+	 * @param limitAdapter the custom limit adopter
+	 * @return a Bulkhead instance
+	 */
+	static AdaptiveBulkhead of(String name, AdaptiveBulkheadConfig config, LimitPolicy limitAdapter) {
+		return AdaptiveLimitBulkhead.factory().createAdaptiveLimitBulkhead(name, config, limitAdapter);
+	}
+
+	/**
+	 * Creates a bulkhead with a custom configuration
+	 *
+	 * @param name   the name of the bulkhead
+	 * @param config a custom BulkheadConfig configuration
+	 * @return a Bulkhead instance
+	 */
+	static AdaptiveBulkhead of(String name, AdaptiveBulkheadConfig config) {
+		return AdaptiveLimitBulkhead.factory().createAdaptiveLimitBulkhead(name, config);
+	}
+
+	/**
+	 * Creates a bulkhead with a custom configuration
+	 *
+	 * @param name                   the name of the bulkhead
+	 * @param bulkheadConfigSupplier custom configuration supplier
+	 * @return a Bulkhead instance
+	 */
+	static AdaptiveBulkhead of(String name, Supplier<AdaptiveBulkheadConfig> bulkheadConfigSupplier) {
+		return AdaptiveLimitBulkhead.factory().createAdaptiveLimitBulkhead(name, bulkheadConfigSupplier.get());
+	}
+
+	interface AdaptiveMetrics extends Bulkhead.Metrics {
```

Comment:
We added `getNumberOfSlowSuccessfulCalls` to `CircuitBreaker.Metrics`C. Can we add this here as well?

Reply:
I guess the user can easily get that if he want to have it by getNumberOfSlowCalls()-getNumberOfFailedCalls() , so i do not feel it is needed as mainly he will be concerned about slow calls in general i guess